codersdk/notifications.go

```diff
@@ -280,3 +281,38 @@ func (c *Client) PostTestWebpushMessage(ctx context.Context) error {
 	}
 	return nil
 }
+
+type CustomNotificationContent struct {
+	Title   string `json:"title"`
+	Message string `json:"message"`
+}
+
+type CustomNotificationRequest struct {
+	Content *CustomNotificationContent `json:"content"`
+	// TODO(ssncferreira): Add target (user_ids, roles) to support multi-user and role-based delivery.
+	//   See: https://github.com/coder/coder/issues/19768
+}
+
+func (c CustomNotificationRequest) Validate() error {
+	if c.Content == nil {
+		return xerrors.Errorf("content is required")
+	}
+	if strings.TrimSpace(c.Content.Title) == "" &&
+		strings.TrimSpace(c.Content.Message) == "" {
+		return xerrors.Errorf("provide a non-empty 'content.title' or 'content.message'")
+	}
+	return nil
+}
```

Comment:
Should we also enforce a maximum length on the notification message title / content? I don't think we enforce any limit at the database. Given that this is exposed to users, I think we should enforce some limit, as it's easier to relax an existing restriction than it is to add a new one.